From 38b2d3f7daa27a9cce88b458bb1fbc389f64bdf2 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Thu, 24 Dec 2020 12:56:45 +0100
Subject: [PATCH] host-pip-requirements: add Babel

Sent to upstream: https://github.com/openwrt/packages/pull/14329

One package (reforis-distutils [1]), which is not in this repo and it is required for Turris routers
because of reForis (simplified user interface), which is being developed
depends on Babel host package [2].

It fails with:
ERROR: Could not open requirements file: [Errno 2] No such file or directory: '/foo-omnia/build/feeds/packages/lang/python/host-pip-requirements/Babel.txt'

I tried to remove HOST_PYTHON3_PACKAGE_BUILD_DEPENDS (as Babel is
dependency for the package), then
reforis-distutils is compiled and package is created, but thing is that
if I want to try compile package reForis [3] then it fails with
following output:

ERROR: Could not find a version that satisfies the requirement reforis_distutils (from versions: none)
ERROR: No matching distribution found for reforis_distutils
Traceback (most recent call last):
  File "/foo/mox/build/staging_dir/hostpkg/lib/python3.8/site-packages/setuptools/installer.py", line 128, in fetch_build_egg
    subprocess.check_call(cmd)
  File "/foo/mox/build/staging_dir/target-aarch64_cortex-a53_musl/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/foo/mox/build/staging_dir/hostpkg/bin/python3.8', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/foo/mox/build/tmp/tmp3s9rxjym', '--quiet', '--find-links', 'git+https://gitlab.nic.cz/turris/reforis/reforis-distutils.git#egg=reforis-distutils', 'reforis_distutils']' returned non-zero exit status 1.

No matter if I add python3-babel into reForis dependency.
This was working on OpenWrt 19.07, but there is regression in OpenWrt
master, which I fixed as packages reForis-distutils and reForis are
compiled successfully with this commit.

[1] https://gitlab.nic.cz/turris/turris-os-packages/-/blob/v5.1.4/web/reforis/reforis-distutils/Makefile#L23
[2] https://gitlab.nic.cz/turris/reforis/reforis-distutils/-/blob/v0.1.0/setup.py#L13
[3] https://gitlab.nic.cz/turris/turris-os-packages/-/blob/v5.1.4/web/reforis/reforis/Makefile

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 lang/python/host-pip-requirements/Babel.txt | 3 +++
 1 file changed, 3 insertions(+)
 create mode 100644 lang/python/host-pip-requirements/Babel.txt

diff --git a/lang/python/host-pip-requirements/Babel.txt b/lang/python/host-pip-requirements/Babel.txt
new file mode 100644
index 000000000..b0a787d17
--- /dev/null
+++ b/lang/python/host-pip-requirements/Babel.txt
@@ -0,0 +1,3 @@
+Babel==2.9.0 --hash=sha256:da031ab54472314f210b0adcff1588ee5d1d1d0ba4dbd07b94dba82bde791e05
+pytz==2020.4 --hash=sha256:3e6b7dd2d1e0a59084bcee14a17af60c5c562cdc16d828e8eba2e683d3a7e268
+
-- 
2.25.1

